samples/mqtt5/mqtt5_pubsub/README.md

```diff
@@ -69,3 +69,233 @@ You can also pass a Certificate Authority file (CA) if your certificate and key
+
+
+### MQTT over Websockets with Cognito
+
+A MQTT5 websocket connection can be made using Cognito to authenticate rather than the AWS credentials located on the device or via key and certificate. Instead, Cognito can authenticate the connection using a valid Cognito identity ID. This requires a valid Cognito identity ID, which can be retrieved from a Cognito identity pool. A Cognito identity pool can be created from the AWS console.
+
+To create a MQTT5 builder configured for this connection, see the following code:
+
+```cpp
+    // Create websocket configuration
+    Aws::Crt::Auth::CredentialsProviderChainDefaultConfig defaultConfig;
+    std::shared_ptr<Aws::Crt::Auth::ICredentialsProvider> provider = Aws::Crt::Auth::CredentialsProvider::CreateCredentialsProviderChainDefault(defaultConfig);
+
+    Aws::Crt::Auth::CredentialsProviderCognitoConfig cognitoConfig;
+    // See https://docs.aws.amazon.com/general/latest/gr/cognito_identity.html for Cognito endpoints
+    cognitoConfig.Endpoint = "cognito-identity.<region>.amazonaws.com";
+    cognitoConfig.Identity = "<Cognito Identity ID>";
+    Aws::Crt::Io::TlsContextOptions tlsCtxOptions = Aws::Crt::Io::TlsContextOptions::InitDefaultClient();
+    cognitoConfig.TlsCtx = Aws::Crt::Io::TlsContext(tlsCtxOptions, Aws::Crt::Io::TlsMode::CLIENT);
+    std::shared_ptr<Aws::Crt::Auth::ICredentialsProvider> provider = Aws::Crt::Auth::CredentialsProvider::CreateCredentialsProviderCognito(cognitoConfig);
+
+    if (!provider)
+    {
+        fprintf(stderr, "Failure to create credentials provider!\n");
+        exit(-1);
+    }
+    Aws::Iot::WebsocketConfig websocketConfig(<signing region>, provider);
+
+    // Create a Client using Mqtt5ClientBuilder
+    Aws::Iot::Mqtt5ClientBuilder *builder = Aws::Iot::Mqtt5ClientBuilder::NewMqtt5ClientBuilderWithWebsocket(
+        "<clientEndpoint>", websocketConfig);
+
+    /* You can setup other client options and lifecycle event callbacks before call builder->Build().
+    ** Once the the client get built, you could no longer update the client options or connection options
+    ** on the created client.
+    */
+
+    // Build Mqtt5Client
+    std::shared_ptr<Aws::Crt::Mqtt5Client> mqtt5Client = builder->Build();
+
+    if (mqtt5Client == nullptr)
+    {
+        fprintf(stdout, "Client creation failed.\n");
+        return -1;
+    }
+
+
+```
+
+**Note**: A Cognito identity ID is different from a Cognito identity pool ID and trying to connect with a Cognito identity pool ID will not work. If you are unable to connect, make sure you are passing a Cognito identity ID rather than a Cognito identity pool ID.
+
+
+### HTTP Proxy
```

Comment:
I think that adds a little bit of complexity to the basic pubsub sample that we should leave out unless someone is specifically looking to use a proxy. It's hard and debatable where the line should be drawn but if we add this, it's debatable that we could also add other additional checks for various other connection types into the basic pubsub. I'm not dead set on keeping it out, but I'd want to collect other opinions on potentially adding it to our basic connect pubsub sample.

Reply:
The "HTTP Proxy" section kind of stands out, because it's not a standalone connection method, but rather an additional connection configuration. So, I started thinking if it's possible to put it aside, and adding it into code is a solution I came up with.
I'm totally fine if we decide to keep it as a separate section, though. Because you're right, it adds complexity to code.

Reply:
I can move it somewhere that makes more sense as an additional connection option rather than a connection method

Reply:
Hm, I couldn't figure out where to put this section. If you'll find a way, that'd be perfect